Update authz docs (#1863)
wso2/docs-choreo-dev: en/developer-docs/docs/administer/control-access-in-the-choreo-console.md

# Control Access in the Choreo Console

> In Choreo, administrators can control user access to different projects and environments within the organization. 

Reviewer: ```suggestion
With Choreo, administrators can control user access to different projects and environments within the organization. 
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

At the finest granularity level, an administrator can restrict a user to perform a specific action on a specific project and a specific environment.

## Access Control (Authorization) Concepts

To understand how to configure access control in Choreo, it is important to understand how access control is modelled in Choreo.

### Fundamental Question in Access Control

> Access control of any system boils down to answering the following question.

Reviewer: ```suggestion
Access control in any system ultimately comes down to answering the following question.
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

> > *Can this **user** perform this **action** on this **resource**?*

Reviewer: WDYT about having this inside an admonition rather than a quote?

Eg:

<img width="1317" height="399" alt="Image" src="https://github.com/user-attachments/assets/edb2a2b7-9592-4376-927b-c3f23613985d" />

Author: Lets keep it as it is as discussed offline

???+ example
    Can Harry view logs of development environment of engineering project? 
    
    - `Harry` is the **user**  
    - `view logs` is the **action**  
    - `development environment of engineering project` is the **resource**   

### Permission

In Choreo access control model, a permission is the right to perform a specific action. 

???+ example

>     A user can configure a custom domain only if the user has `choreo:domain_manage` permission.

Reviewer 2: Can we change this to a commonly used permission like - `create component` or something

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

#### Environment Specific Permissions

> In Choreo, a subset of actions such as `deploy component`, `promote component`, `create configuration groups`, and `view logs` are always performed within the context of an environment. The permissions which give the right to perform such actions are categorized as **Environment Specific** permissions.

Reviewer: ```suggestion

In Choreo, actions such as `deploy component`, `promote component`, `create configuration groups`, and `view logs` must be performed within the context of an environment. The permissions which give the right to perform such actions are categorized as **Environment Specific** permissions.
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

!!! note

>     For the actions which are tied to an environment, in some cases, it makes sense to allow a user to perform the action on some environments but not on other environments. For example, a developer should be allowed to view logs of the develoment environment. But the same developer should not be allowed to view logs of the production environment.

Reviewer: ```suggestion
    For actions tied to an environment, it is sometimes necessary to allow a user to perform the action in certain environments but not in others; for example, a developer may be allowed to view logs in the development environment but not in the production environment.
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

### Role

In practice, assigning permissions individually to each user is a time consuming and error prone task. Moreover, 

> users with similar job responsibilities often need the same set of permissions. Roles are collections of such permissions. They are designed to match real-world job responsibilities.   

Reviewer: ```suggestion
Assigning permissions individually to each user is a time-consuming and error prone task. Users with similar job responsibilities often need the same set of permissions, which is why **roles** exist. A role is a collection of permissions designed to reflect real-world job responsibilities.
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

> Roles simplify assigning permissions to users. Instead of assigning 50 developers 20 permissions each (1000 assignments), you just assign the 20 permissions to the `Developer` role and assign the `Developer` role to each developer (20 + 50 assignments).  

Reviewer: ```suggestion
Roles simplify permission management. For example, instead of assigning 20 permissions to each of 50 developers (1000 assignments), you can assign the 20 permissions to a Developer role and then assign that role to each developer (20 + 50 assignments).
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)

!!! info

>     Each organization in Choreo has a predefined set of roles with permissions assigned to them by default. Organization adminstratrators can further customize the existing roles and and create new roles as required. [Learn more](../choreo-concepts/organization.md#roles)

Reviewer: ```suggestion
    Each organization in Choreo comes with a predefined set of roles with default permissions. Organization administrators can customize existing roles or create new ones as needed. [Learn more](../choreo-concepts/organization.md#roles)
```

Author: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1863/commits/8b51bec7f34bcbee751f3c961aa216cdc7fb00f5)
